Add multi select pattern documentation (#1938)
hashicorp/design-system: website/docs/components/table/partials/guidelines/guidelines.md

![Example of multi-select within table cells](/assets/components/table/multi-select-cells.png)

!!! Info

> For more details around using a multi-select Table, recommended patterns, and intended interactions visit the [Multi-select patterns](/patterns/multi-select-patterns) documentation.

Reviewer: ```suggestion
For more details around using a multi-select Table, recommended patterns, and intended interactions visit the [Multi-select patterns](/patterns/table-multi-select) documentation.
```

Author: Shoot, nice catch on this.

> For more details around how to approach a multi-select Table, recommended patterns, and intended interactions visit the [Multi-select patterns](/patterns/multi-select-patterns) documentation.

Reviewer 2: ```suggestion
For more details around using a multi-select Table, recommended patterns, and intended interactions visit the [Multi-select patterns](/patterns/multi-select-patterns) documentation.
```

![Example of multi-select within table cells](/assets/components/table/multi-select-cells.png)

!!! Info

> For more details around how to approach a multi-select Table, recommended patterns, and intended interactions visit the [Multi-Selection patterns](/patterns/multi-selection) documentation.

Reviewer 2: Don't forget to update the URL in the above as well.

"/patterns/multi-selection"
=>
"/patterns/multi-select"

!!!

> <!--@jory move this to the pattern documentation-->

Author: Yes, thanks for catching this, removed it
